modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,32 +109,53 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+      (@validate_platform ? is_module_platform?(mod) : true) &&
+      (@validate_arch ? is_module_arch?(mod) : true) &&
       is_module_options_ready?(mod)
-    )
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    setup_validation_options
+    setup_color_options
+
     # Collects exploits into an array
     @local_exploits = []
-    framework.exploits.each do |name, _obj|
+    framework.exploits.each_with_index do |(name, _obj), index|
+      @valid_reasons = []
+
+      print "%bld%blu[*]%clr Gathering exploit #{index + 1} / #{framework.exploits.count}\r"
       mod = framework.exploits.create name
       next unless mod
+
       set_module_options mod
-      next unless is_module_wanted? mod
-      @local_exploits << mod
+
+      @local_exploits << { module: mod, wanted: true } if is_module_wanted? mod
+      @local_exploits << { module: mod, wanted: false, valid_reasons: @valid_reasons } if @valid_reasons.any?
```

Comment:
Potentially this could just store the raw data:
```suggestion
      @local_exploits << { module: mod, result: verify_mod(mod) }
```

And then `verify_mod` (name to be confirmed) would do similar to what you have in the old predicate methods - but now without the side effects:

```ruby
def verify_mod(mod)
   return { check: false } unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
   {
      incompatibility_reasons: session_incompatibility_reasons(mod),
      is_module_platform: is_module_platform?(mod),
      is_module_arch: is_module_arch?(mod),
      is_module_options_ready: is_module_options_ready?(mod)
   }
end
```

You can still keep an implementation of `is_module_wanted?` around, but it would make use of the verify mod result instead.